Tutorial12.ipynb

```diff
@@ -0,0 +1,374 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Tutorial 12\n",
+    "## A How-to Approach of Maximum Likelihood and Regular Expressions"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "import pandas as pd\n",
+    "import numpy as np\n",
+    "import os\n",
+    "from scipy.optimize import minimize\n",
+    "from scipy.stats import norm\n",
+    "import scipy.stats\n",
+    "\n",
+    "os.chdir('/Users/andrewguinness/Google Drive/ND/Fall 17/Intro to Biocomputing/Intro_Biocom_ND_319_Tutorial12')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Let's begin by importing all of the packages we will need in a manner conducive to their use, and set our working directory to the directory containing our data. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "chickwts = pd.read_csv(\"chickwts.txt\")\n",
+    "\n",
+    "chickwts = chickwts[(chickwts.feed == 'soybean') | (chickwts.feed == 'sunflower')]\n",
+    "\n",
+    "chickwts = chickwts.replace('soybean', 0)\n",
+    "chickwts = chickwts.replace('sunflower',1)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now we want to read in our data and save it as an argument. \n",
+    "\n",
+    "\n",
+    "Thereafter, we want to subset our data so that we can answer the question we're asking **statistically**. So, let's grab the chick weights **ONLY** for those fed soybean and sunflower diets. (Read the | bar as \"or\"). \n",
+    "\n",
+    "Now, let's treat soybean like our intercept of the linear model, b0 by setting it to zero. \n",
+    "\n",
+    "What this does is it takes the linear model: y = b0 + b1 * x and changes it to y = b0 (because x = 0). \n",
+    "\n",
+    "This will become clearer later. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Optimization terminated successfully.\n",
+      "         Current function value: 138.469162\n",
+      "         Iterations: 200\n",
+      "         Function evaluations: 363\n",
+      "[ 246.42855057   82.48813575   49.73948886]\n"
+     ]
+    },
+    {
+     "data": {
+      "text/plain": [
+       "Ttest_indResult(statistic=array([-4.05020659,        -inf]), pvalue=array([ 0.00046409,  0.        ]))"
+      ]
+     },
```

Comment:
Using t-test is a good way to go. Good job. If you use likelihood test, you want to use both null hypothesis and alternative model